### ISSUE TITLE
clone must not change validity

### ISSUE THREAD

**Issue by fingerpich** [fingerpich/jalali-moment#29]
The following code shows the problem
```js
const m = jmoment('hello', 'jMM/jDD/jYYYY', 'fa');
const isMom = jmoment.isMoment(m); // true
const cloned = m.clone().locale('fa');
const v1 = m.isValid(); // false​ it s ok​
const v2 = cloned.isValid(); // true?? why is that​
```

**Comment by peymanebrahimi** [comment]
hi tnx
do we need a new version?

**Comment by fingerpich** [comment]
Yes, Its fixed in version 2.1.4 that I have just released.

**Comment by peymanebrahimi** [comment]
const result = jmoment();
` ERROR TypeError: Cannot read property '_isAMomentObject' of undefined `
refers to ` if (config._isValid === false || (input._isAMomentObject && !input._isValid)){ ` in library.
Would be nice if there were a test for:
```
    const year = 1396;
    const month = 0;
    const date = 1;
    const result = jmoment().jYear(year).jMonth(month).jDate(date)
    .hours(0).minutes(0).seconds(0).milliseconds(0).locale('fa');
```
